tsdb/engine/tsm1/reader.go

```diff
@@ -607,7 +607,7 @@ func (d *indirectIndex) UnmarshalBinary(b []byte) error {
 		i++
 
 		// 2 byte count of index entries
-		count := btou16(b[i : i+indexCountSize])
+		count := int32(btou16(b[i : i+indexCountSize]))
```

Comment:
I must be dense, but I don't see how this affects anything. Where exactly does a 16-bit wide count affect anything?